cli/file_fetcher.rs

```diff
@@ -1721,6 +1777,7 @@ mod tests {
     let client = HttpClient::from_client(
       create_http_client(
         version::get_user_agent(),
+        None,
```

Comment:
Add a separate method next to `create_http_client`, since most tests won't use the custom resolver.